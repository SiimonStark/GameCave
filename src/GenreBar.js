import React, { Component } from 'react';
import CheckBox from './CheckBox.js';

export default class GenreBar extends Component {
    constructor(props) {
        super(props);
        this.state = {showGenres: false, mouseIn: false}
    }
    toggleMouse = () => {
        this.setState({ mouseIn: this.state.mouseIn ? false : true });
    }
    focusLost = () => {
        this.setState({ showGenres: this.state.mouseIn ? true : false });
    }
    toggleGenres = () => {
        this.setState({ showGenres: this.state.showGenres ? false : true });
    }
    render() {
        let hideGenres = this.state.showGenres ? "genre-check" : "genre-check hide";
        let buttonSelector = this.props.genresToFilter.length ? "button current" : "button";
        let genres = this.props.genres.sort();
        return (
            <div onBlur={this.focusLost} onMouseEnter={this.toggleMouse} onMouseLeave={this.toggleMouse} className="genre-bar search-section">
                <form onBlur={this.focusLost} tabIndex="0" className={hideGenres}>
                    {
                        genres.map((val, i) => {
                            return <CheckBox name={val}
                                updateChecks={this.props.updateChecks}
                                class="genresToFilter check-box"
                                key={i} />
                        })
                    }
                </form>
                <button onClick={this.toggleGenres} className={buttonSelector}>GENRES</button>
            </div>
        )
    }
}