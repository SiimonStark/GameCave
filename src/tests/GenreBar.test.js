import React from 'react';
import ReactDOM from 'react-dom';
import GenreBar from '../GenreBar';
import { shallow } from 'enzyme';

const mockGenres = ['Action', 'Adventure', 'Fantasy']
const mockFilters = ['Action', 'Adventure']

describe('GenreBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <GenreBar 
        genres={mockGenres}
        genresToFilter={mockFilters}
      />
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      showGenres: false, mouseIn: false
    });
  })

  it('should toggle mouseIn on mouse leave', () => {
    expect(wrapper.state('mouseIn')).toEqual(false);
    wrapper.find('.genre-bar').simulate('mouseleave');
    expect(wrapper.state('mouseIn')).toEqual(true);
  })

  it('should toggle mouseIn on mouse enter', () => {
    expect(wrapper.state('mouseIn')).toEqual(false);
    wrapper.find('.genre-bar').simulate('mouseleave');
    expect(wrapper.state('mouseIn')).toEqual(true);
    wrapper.find('.genre-bar').simulate('mouseenter');
    expect(wrapper.state('mouseIn')).toEqual(false);
  })

  it('should toggle showGenres when button is clicked', () => {
    expect(wrapper.state('showGenres')).toEqual(false);
    wrapper.find('.button').simulate('click');
    expect(wrapper.state('showGenres')).toEqual(true);
  }) 

  it('should change showGenres on blur', () => {
    wrapper.find('.button').simulate('click');
    expect(wrapper.state('showGenres')).toEqual(true);
    wrapper.find('.genre-bar').simulate('blur');
    expect(wrapper.state('showGenres')).toEqual(false);
  })
})