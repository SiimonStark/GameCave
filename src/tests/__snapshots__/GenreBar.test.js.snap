// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreBar should match snapshop 1`] = `
<div
  className="genre-bar search-section"
  onBlur={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <form
    className="genre-check hide"
    onBlur={[Function]}
    tabIndex="0"
  >
    <CheckBox
      class="genresToFilter check-box"
      key="0"
      name="Action"
    />
    <CheckBox
      class="genresToFilter check-box"
      key="1"
      name="Adventure"
    />
    <CheckBox
      class="genresToFilter check-box"
      key="2"
      name="Fantasy"
    />
  </form>
  <button
    className="button current"
    onClick={[Function]}
  >
    GENRES
  </button>
</div>
`;
